File: src/main/java/com/learning/java/controlstruct/ConditionalsSyntax.java
package com.learning.java.controlstruct;

/* Java: cинтаксис условных конструкций */


public class ConditionalsSyntax {

	public static void main(String[] args) {
		/* (#1)  if:then - выполняется условное действие в зависимости от состояния */
		if (args.length==0)
			System.out.println("#1: Arguments are null");
		else
			System.out.println("#1: Arguments are not null");
		
		/* (#2)  ?: - более короткая форма выражения условного действия (if:then) */
		System.out.println(args.length==0 ? "#2: Arguments are null" : "#2: Arguments are not null");
		
		/* (#3) switch:case - список состояний для условный */
		String day = "Monday";
		String dayType = "???"; // (текущий день недели..)
		switch(day) {
			case "Monday":
			case "Tuesday":
			case "Wednesday":
			case "Thursday":
			case "Friday":
				dayType = "working day";
				break;
			case "Saturday":
			case "Sunday":
				dayType = "holiday";
				break;
			default:
				dayType = "unknown";
				break;
		}
		System.out.println("#3: " + day + " is " + dayType);
	}
}

File: src/main/scala/com/learning/scala/controlstruct/ConditionalsSyntax.scala
package com.learning.scala.controlstruct

/* Scala: cинтаксис условных конструкций */


object ConditionalsSyntax { 
  
  def main(args: Array[String]) {
    /* (#1)  if:then - возвращает условное значение в зависимости от состояния = условные выражения в Scala эквивалентно (?:) */
    println( if(args.length==0) "#1: Arguments are null" else "#1: Arguments are not null" )
    
    /* (#2)  ?: */
    /* даже любое тело в Scala является выражением {...} и может вернуть что-то */

    /* (#3)  match:case - Scala не имеет никакого (switch:case) вместо этого она использует выражения 'Match / Pattern Matching' */
    val day = "Monday"
    println("#3: " + day + " is " +
      (day match {
        case "Monday" | "Tuesday" | "Wednesday" | "Thurday" | "Friday" => "working day"
        case "Saturday" | "Sunday" => "holiday"
        case _ => "unknown"
      })
    )
  }
}
